### ISSUE TITLE
Rule idea: at-media-no-non-variable / at-media-no-non-dollar-variable

### ISSUE THREAD

**Issue by dryoma** [kristerkari/stylelint-scss#29]
... or something like this. The point is to force all values inside `@media` queries be variables. The `-dollar-` variant is to underline that the rule forces `$var`s, not custom properties or variables from some other syntaxes.

Examples:
```scss
// Good:
@media (max-width: $width-mid) { ... }

// Bad:
@media (max-width: 1024px) { ... }
```

**Comment by kristerkari** [comment]
Why not, could be useful. 👍

**Comment by dryoma** [comment]
I have some cases for this rule I'd like to discuss. Suppose we have it as `media-query-value-dollar-variable: "always"/"never"`, which by itself looks ok. Then:

```scss
@media (max-width: 1024px) { ... } // Warning for "always"
@media (max-width: $var) { ... } // Warning for "never"

@media (max-width: 1024px + $var) { ... } // False positive for "never"
@media (max-width: $var + 1024px) { ... } // Not even sure how to treat this

@media (max-width: someFunc()) { ... } // Same here
```

1) We might need some additional rules for `calculation` and `function` as a media query value. 2) Or just process those in this rule when using some additional options like `ignoreCalculation`. The first option seems decent at first, but not sure at all about a rule just prohibiting a math operation in a value. The second one would keep it all compact, but might result in too many options.

What woul be your thoughts guys?

**Comment by kristerkari** [comment]
My opinion is that if the rule is called `media-query-value-dollar-variable`, then it should only care about dollar variables, so in my opinion:
- For `always` only allow `$var` and `#{$var}` (because it would be difficult to allow any kind of calculation with all kinds of edge cases).
- For `never` don't allow any `$var` or calculation that contains `$var`, but allow calculation like `1000px + 24px` because it does not contain a variable.

I just noticed when writing this that that is exactly how you have implemented that rule, so 👍

**Comment by kristerkari** [comment]
Implemented in https://github.com/kristerkari/stylelint-scss/pull/47
